tests/Unit/UserTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Tests\Unit;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithoutMiddleware;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use App\Models\User;
+use App\Models\Order;
+
+class UserTest extends TestCase
+{
+    /**
+     * A basic test example.
+     *
+     * @return void
+     */
+    public function testExample()
+    {
+        $this->assertTrue(true);
+    }
+
+    public function test_contains_valid_fillable_properties()
+    {
+        $m = new User();
+        $this->assertEquals(['name', 'email', 'password', 'phone_number', 'gender', 'role', 'address', 'avatar'], $m->getFillable());
+        $this->assertEquals('id', $m->getKeyName());
+    }
+
+    public function test_user_relation()
+    {
+        $m = new Order();
+        $relation = $m->user();
+        $this->assertEquals('user_id', $relation->getForeignKey());
+        $this->assertEquals('id', $relation->getOwnerKey());
```

Comment:
có những loại assert nào em

Reply:
Dạ theo em tìm hiểu thì được dùng nhiều là
assertEquals và assertTrue
assertEquals nếu 2 chuỗi equals thì sẽ pass testcase. Còn nếu 2 chuỗi not equals nó sẽ fail testcase và throw ra message.
assertTrue thì sẽ trả về true false. Nếu true thì sẽ pass testcase, ngược lại sẽ fail và throw ra AssertionError

Như trên em làm là assertEquals(String actual,String expected) nếu nó verify 2 chuỗi string arguments là giống nhau thì sẽ trả về là pass testcase.

Còn những loại khác em đang tìm hiểu như là assertViewHasAll, assertViewHasAll, assertRedirectedTo, assertStatus, ... thì em đang tìm hiểu cách dùng và ứng dụng bên controller ạ